### ISSUE TITLE
BPF: Map migration doesn't clean stale :pending maps before renaming another map to that

### ISSUE THREAD

**Issue by gentoo-root** [cilium/cilium#24013]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

1. Deploy Cilium with a broken BPF program that will fail the verifier and cause a restart loop. (That might be a good Cilium as well, but the timing needs to be right to run `cilium uninstall` in step 2 before the map migration finishes.)
2. Deploy known good Cilium to the same cluster using `cilium uninstall` and `cilium install ...`.
3. It seems that if `cilium uninstall` happens during a migration of a BPF map, a leftover map with `:pending` suffix may remain.
4. Installing good Cilium fails, because it attempts to do a migration of that map, but when it tries to rename it to append the `:pending` suffix, it fails, because the stale map already exists. (`cilium install` actually reports a success, but the subsequent `cilium status` prints errors.)

Cilium should check for any existing `:pending` maps and clear them before trying to rename to `:pending`.

### Cilium Version

Current master (9755cde295495289cbab6c3763fc0bf10fdbec9f)

### Kernel Version

```
Linux gke-max-cluster-default-pool-dee88c1f-oiqu 5.15.65+ #1 SMP Fri Dec 30 10:10:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Kubernetes Version

```
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"archive", BuildDate:"2023-01-18T22:38:55Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6-gke.200", GitCommit:"6182692c33818a2b5998c4ca45b2a026b6ac9f97", GitTreeState:"clean", BuildDate:"2023-01-20T09:30:01Z", GoVersion:"go1.19.5 X:boringcrypto", Compiler:"gc", Platform:"linux/amd64"}
```

### Sysdump

[cilium-sysdump-20230224-160125.zip](https://github.com/cilium/cilium/files/10824910/cilium-sysdump-20230224-160125.zip)

### Anything else?

[cilium.log](https://github.com/cilium/cilium/files/10824926/cilium.log)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by margamanterola** [comment]
After discussing with Maxim, moving this to sig/agent, as it seems that it's part of the Agent responsibilities to clean up these maps.

**Comment by ti-mo** [comment]
This has been addressed by https://github.com/cilium/cilium/pull/28740/files#diff-3e9e0ea0e6ba89364beb4c2dcd53017e55ff1e95a7a8b6b68345223e81c2b03fR121 and has been released in 1.14.4 and 1.13.9.

For context, moving an existing map pin will now always overwrite an existing (older) `:pending` copy. This is an unfortunate consequence of the migration mechanism. There's no point in being defensive and bailing out if a :pending map already exists; we don't know which one is good or bad. Since we're no longer using iproute2 to load BPF, we don't need migrations anymore; we can simply overwrite pins after an ELF's entrypoints were successfully attached. I've created https://github.com/cilium/cilium/issues/29333 to this effect.
